File: LeetCode/3691-minimum-operations-to-make-columns-strictly-increasing/minimum-operations-to-make-columns-strictly-increasing.py
class Solution:
    def minimumOperations(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        op = 0

        for j in range(n):
            for i in range(1, m):
                if grid[i][j] <= grid[i-1][j]:
                    inc = grid[i-1][j] - grid[i][j] + 1;
                    grid[i][j] += inc;
                    op += inc;
        
        return op;
        

File: LeetCode/3691-minimum-operations-to-make-columns-strictly-increasing/minimum-operations-to-make-columns-strictly-increasing.cpp
class Solution {
public:
    int minimumOperations(vector<vector<int>>& grid) {
        int n = grid.size();
        int m = grid[0].size();

        int op = 0;
        for(int j = 0; j < m; j++){
            for(int i = 1; i < n; i++){
                if(grid[i][j] <= grid[i-1][j]){
                    int inc = grid[i-1][j] - grid[i][j] + 1;
                    grid[i][j] += inc;
                    op += inc;
                }
            }
        }

        return op;
    }
};